src/DynamoCoreWpf/ViewModels/Core/PortViewModel.cs

```diff
@@ -511,49 +512,120 @@ private void AutoCompleteCluster(object parameter)
+                        var sourcePort = sourceNode.OutPorts.FirstOrDefault(p => p.PortModel.Index == connection.StartNode.PortIndex - 1);
+                        var targetPort = targetNode.InPorts.FirstOrDefault(p => p.PortModel.Index == connection.EndNode.PortIndex - 1);
+                        var commands = new List<DynamoModel.ModelBasedRecordableCommand>
+                        {
+                            new DynamoModel.MakeConnectionCommand(sourceNode.Id.ToString(), connection.StartNode.PortIndex - 1, PortType.Output, DynamoModel.MakeConnectionCommand.Mode.Begin),
+                            new DynamoModel.MakeConnectionCommand(targetNode.Id.ToString(), connection.EndNode.PortIndex - 1, PortType.Input, DynamoModel.MakeConnectionCommand.Mode.End),
+                        };
+                        commands.ForEach(c =>
+                        {
+                            try
+                            {
+                                wsViewModel.DynamoViewModel.Model.ExecuteCommand(c);
+                            }
+                            catch (Exception) { }
+                        });
+                    });
+
+                    // Connect the cluster to the original node and port
+                    var finalCommands = new List<DynamoModel.ModelBasedRecordableCommand>
+                    {
+                        new DynamoModel.MakeConnectionCommand(node.Id.ToString(), 0, PortType.Output, DynamoModel.MakeConnectionCommand.Mode.Begin),
+                        new DynamoModel.MakeConnectionCommand(targetNodeFromCluster?.Id.ToString(), ClusterResultItem.EntryNodeInPort, PortType.Input, DynamoModel.MakeConnectionCommand.Mode.End),
+                    };
+                    finalCommands.ForEach(c =>
+                    {
+                        try
+                        {
+                            wsViewModel.DynamoViewModel.Model.ExecuteCommand(c);
+                        }
+                        catch (Exception) { }
+                    });
+
+                    // AutoLayout should be called after all nodes are connected
+                    foreach(var node in clusterMapping.Values)
+                    {
+                        wsViewModel.DynamoViewModel.Model.AddToSelection(node.NodeModel);
+                    }
+                    wsViewModel.Model.DoGraphAutoLayout(false, true, node.Id);
+                }
+
+                // Display the cluster info in the right side panel
+                // wsViewModel.OnRequestNodeAutoCompleteViewExtension(results);
+            }
+            catch (Exception)
+            {
+                // Log the exception and show a notification to the user
+            }
+        }
+
+        /// <summary>
+        /// Create a mock cluster. This is test only and should be removed when Cluster AutoComplete is in production
+        /// </summary>
+        private void CreateMockCluster()
```

Comment:
Do we still need this ?

Reply:
I can delete in another PR after https://jira.autodesk.com/browse/DYN-8358 closed, I was thinking this might be useful for testing